### PR TITLE
Natural sorting

### PR DESCRIPTION
**Pull request by JacksonCampolattaro**
Check-list
----------

 * [x] There is a linked issue discussing the motivations for this feature or bugfix (see: #3627)
 * [ ] Unit tests have been added where possible
 * [ ] I've added / updated documentation for any user-facing features.
 * [ ] Performance seems to be comparable or better than current `master`

What this change is adding / fixing
-----------------------------------
Replaces the sorting function used by the file system browser, so that directories appear in a more intuitive order:
```
Example5/
example1/
example10
example2/
```
becomes
```
example1/
example2/
Example5/
example10
```

Next Steps
---------------
The current solution is functional, but there are a few additional changes that could be included:
- Apply the same sorting function to other Gtk.TreeViews (e.g. Playlists or Songs could be sorted naturally when put in alphabetical order)
- Add an option to enable or disable this behavior
- Add a context menu to quickly switch between different sorting approaches (natural, ascii order, case-ignoring) by right clicking on the column header
- Benchmark performance on very large directory structures (natural sorting uses regex and integer parsing, so it definitely comes with at least a small performance penalty)

### CODE REVIEW

**Comment by LAlbani98** [comment]
With reference to the natural-sorting branch involved in this #3948 pull-request, [i added a simple menu](https://github.com/LAlbani98/quodlibet/blob/natural-sorting-menu/quodlibet/qltk/filesel.py) allowing to change sorting between ASCENDING and DESCENDING order based upon the compare function designed by @JacksonCampolattaro.
Other sorting orders and model may be added with ease.

File: quodlibet/qltk/filesel.py [hunk 1]
```diff
@@ -191,6 +193,50 @@ def __init__(self, initial=None, folders=None):
...
+
+        def compare(m: Gtk.TreeModel, a: Gtk.TreeIter, b: Gtk.TreeIter, _):
+
+            # Do not sort the top level directories
+            a_path = m.get_path(a)
+            if a_path is not None and a_path.get_depth() == 1:
+                return 0
+
+            @total_ordering
+            class KeyElem:
```
[the remark this answers is not in the document]
Comment (JacksonCampolattaro) [reply quodlibet/qltk/filesel.py:205]: Yeah, there are enough edge cases involved that tests are going to be essential.

[the remark this answers is not in the document]
Comment (declension) [reply quodlibet/qltk/filesel.py:205]: Great - let me know if there's anything here you'd like help with

**Comment by declension** [comment]
Any updates on this?

**Comment by JacksonCampolattaro** [comment]
> Any updates on this?

Hi! Sorry, I've been a little bit sidetracked starting a new job; I might be able to put together some unit tests later this week though.
